process/block/baseProcess.go

```diff
@@ -654,7 +692,13 @@ func (bp *baseProcessor) setProcessingTypeAndConstructionStateForScheduledMb(
 			return err
 		}
 	} else {
-		err = miniBlockHeaderHandler.SetConstructionState(int32(block.Final))
+		constructionState := int32(block.Final)
```

Comment:
you can extract into a method updateConstructionState and reused in the setProcessingTypeAndConstractionStateForNormalMb as well.

Reply:
Done